validator/api/src/main/java/tech/pegasys/teku/validator/api/ValidatorConfig.java

```diff
@@ -243,5 +300,38 @@ private void validateKeyStoreFilesAndPasswordFilesConfig() {
         throw new InvalidConfigurationException(errorMessage);
       }
     }
+
+    private void validateExternalSignerUrlAndPublicKeys() {
+      if (validatorExternalSignerPublicKeys == null
+          || validatorExternalSignerPublicKeys.isEmpty()) {
+        return;
+      }
+
+      if (validatorExternalSignerUrl == null) {
+        final String errorMessage =
+            "Invalid configuration. '--validators-external-signer-url' and '--validators-external-signer-public-keys' must be specified together";
+        throw new InvalidConfigurationException(errorMessage);
+      }
+    }
+
+    private void validateExternalSignerKeystoreAndPasswordFileConfig() {
+      if (validatorExternalSignerKeystore == null
+          && validatorExternalSignerKeystorePasswordFile == null) {
+        return;
+      }
+      final String errorMessage =
+          "Invalid configuration. '--validators-external-signer-keystore' and '--validators-external-signer-keystore-password-file' must be specified together";
+      throw new InvalidConfigurationException(errorMessage);
+    }
```

Comment:
not sure this check is correct either, like `validatorExternalSignerTruststoreAndPasswordFileConfig` seems to require they're both null...

Reply:
let me re-write the condition.

Reply:
@rolfyone 've re-wrote the condition and added unit test case for coverage as well.